core/templates/dev/head/filters.js

```diff
@@ -407,6 +409,21 @@ oppia.filter('stripFormatting', [function() {
   };
 }]);
 
+oppia.filter('getAbbreviatedText', [function () {
```

Comment:
Add more tests -- very long one-word subjects, subjects without spaces, subjects with non-ascii chars. Try to cover all the weird cases that can occur without thinking about how this function is actually implemented.

Reply:
I have added a few more test I think those cover the cases you mentioned.